pontoon/insights/tasks.py

```diff
@@ -0,0 +1,391 @@
+import logging
+
+from celery import shared_task
+from datetime import timedelta
+
+from django.contrib.auth.models import User
+from django.utils import timezone
+
+from pontoon.actionlog.models import ActionLog
+from pontoon.base.models import Entity, Locale, Translation
+from pontoon.base.utils import group_dict_by
+from pontoon.insights.models import LocaleInsightsSnapshot
+
+
+log = logging.getLogger(__name__)
+
+
+@shared_task(bind=True)
+def collect_insights(self):
+    """
+    Gather data needed for the Insights tab and store it in the DB.
+    """
+    start_of_today = (
+        timezone.now()
+        .replace(hour=0)
+        .replace(minute=0)
+        .replace(second=0)
+        .replace(microsecond=0)
+    )
+
+    # Get data sources to retrieve insights from
+    privileged_users = get_privileged_users()
+    contributors = get_contributors()
+    active_users_actions = get_active_users_actions(start_of_today)
+    suggestions = get_suggestions()
+    activity_actions = get_activity_actions(start_of_today)
+    entities = get_entities(start_of_today)
+
+    # Collect insights for each locale
+    snapshots = []
+    for locale in Locale.objects.available():
+        snapshot = get_locale_insights_snapshot(
+            locale,
+            start_of_today,
+            privileged_users[locale.id],
+            contributors[locale.id],
+            active_users_actions[locale.id],
+            suggestions[locale.id],
+            activity_actions[locale.id],
+            entities[locale.id],
+        )
+        snapshots.append(snapshot)
+
+    LocaleInsightsSnapshot.objects.bulk_create(snapshots)
+
+    log.info("Task {}: Locale insights created.".format(self.request.id))
+
+
+def get_privileged_users():
+    """Get managers and translators."""
+    privileged_users = (
+        Locale.objects.available()
+        .prefetch_related("managers_group__user_set", "translators_group__user_set")
+        .values(
+            "pk",
+            "managers_group__user__last_login",
+            "managers_group__user",
+            "translators_group__user",
+        )
+    )
+
+    return group_dict_by(privileged_users, "pk")
+
+
+def get_contributors():
+    """Get all contributors without system users.
+
+    Note that excluding system users directly in the QuerySet is slow.
```

Comment:
I wonder what's the reason behind that issue -> in theory, it shouldn't be a performance problem when you don't use a subquery. 
Is there a difference between `exclude(pk__in=list(excluded_users))` and `exclude(pk__in=excluded_users)`?

Reply:
What I mean to say is that this would be slow:

```python
contributors = (
    Translation.objects
    .filter(user__isnull=False)
    .exclude(user__email__regex=r"^pontoon-(\w+)@example.com$")
    .values("locale", "user")
    .distinct()
)
```

I assume that's because all translations are joined with users.

I simplified the query with your approach.